packages/core/src/contrib/Functions.ts

```diff
@@ -1008,6 +1008,115 @@ export const compDict = {
     returns: valueT("PathCmd"),
   },
 
+  firstPoint: {
+    name: "firstPoint",
+    description: "Returns the first point in a list.",
+    params: [
+      { name: "points", type: realNMT(), description: "list of points" },
+    ],
+    body: (_context: Context, points: ad.Num[][]): VectorV<ad.Num> => {
+      return {
+        tag: "VectorV",
+        contents: points[0],
+      };
+    },
+    returns: valueT("Real2"),
+  },
+
+  lastPoint: {
+    name: "lastPoint",
+    description: "Returns the last point in a list.",
+    params: [
+      { name: "points", type: realNMT(), description: "list of points" },
+    ],
+    body: (_context: Context, points: ad.Num[][]): VectorV<ad.Num> => {
+      return {
+        tag: "VectorV",
+        contents: points[points.length - 1],
+      };
+    },
+    returns: valueT("Real2"),
+  },
+
+  averagePoint: {
```

Comment:
We already have `sumVectors` defined as follows:
```
  sumVectors: {
    name: "sumVectors",
    description: "Return the sum of vectors in a list of vectors.",
    params: [{ name: "vecs", description: "vectors", type: realNMT() }],
    body: (_context: Context, vecs: ad.Num[][]): MayWarn<VectorV<ad.Num>> => {
      if (vecs.length === 0) {
        throw new Error("Expect a non-empty list of vectors");
      }
      const vlen = vecs[0].length;
      const zeros: ad.Num[] = new Array(vlen).fill(0);
      return noWarn(
        vectorV(vecs.reduce((curr, v) => ops.vadd(curr, v), zeros))
      );
    },
    returns: realNT(),
  },
```
Is the `averagePoint` function necessary here?

Reply:
Yes—see my description in the PR. ;-)

Reply:
Yes—see my description in the PR. ;-)

Reply:
Sounds good!